rfc-117-containerised-runtime.md

```diff
@@ -0,0 +1,68 @@
+# RFC 000: Use a containerised runtime
+
+## Summary
+
+Use a containerised runtime for hosting GOV.UK applications. GOV.UK applications currently run on virtual machines running Ubuntu Trusty Tahr. There is currently an extended support contract for Ubuntu Trusty which will end in April 2022. Rather than upgrading Ubuntu on the virtual machines GOV.UK should change the application runtime to a containerised platform.
+
+## Problem
+
+GOV.UK currently hosts the majority of its applications and services using virtual machines running Ubuntu 14.04 LTS (Trusty Tahr).
+
+Trusty is out of date, it was released in April 2014 and standard support ended in April 2019. At this time GOV.UK signed up for [Extended Security Maintenance](https://ubuntu.com/esm) (ESM) which provides continued security fixes for high and critical common vulnerabilities and exposures (CVEs) for the [most commonly used packages](https://wiki.ubuntu.com/SecurityTeam/ESM/14.04#A14.04_Infrastructure_ESM_Packages). ESM will end in April 2022.
+
+Running an old operating system is preventing GOV.UK from upgrading key components of our technology stack (MongoDB, Puppet, Python, ...).
```

Comment:
We're running MongoDB 3.2 on Ubuntu, as well as 2.4, so I'm not sure the reason why we're stuck using 2.4 for many apps is because of using an older version of Ubuntu (although I realise this is a factor in general).

Reply:
Mongo 4.2 claims to [drop support for Trusty](https://docs.mongodb.com/manual/tutorial/install-mongodb-on-ubuntu/#platform-support). Not that we necessarily _want_ to upgrade to Mongo 4.2 - we might choose to move to managed Mongo like DocumentDB, which would avoid this. Still though, being stuck on Trusty is narrowing our options.